rules/linux/persistence_rc_script_creation.toml

```
@@ -0,0 +1,89 @@
+- /lib/systemd/system/rc-local.service
+- /run/systemd/generator/multi-user.target.wants/rc-local.service
+
+In case the file is not present here, the `sudo systemctl status rc-local` can be executed to find the location of the rc-local unit file. Make sure to investigate all files mentioned above, and files that these scripts may link to establish whether the alert is malicious or benign behavior.
+
+### Investigating RC script execution
+The detection rule queries for the creation of these files, but manual analysis is required to check for rc script execution. Systemd will generate syslogs in case of the execution of the rc-local service. The following command can be used to check for the execution of this service:
+
+`sudo cat /var/log/syslog | grep "rc-local.service|/etc/rc.local Compatibility"`
+
+If logging is found, analyze it, and chances are that the contents of the rc.local file have been executed. In case several syslog log files are available, use a wildcard to search through all of the available logs.
+
+### Response and remediation
+- Initiate the incident response process based on the outcome of the triage.
+- Isolate the involved host to prevent further post-compromise behavior.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are identified. Reset passwords for these accounts and other potentially compromised credentials, such as email, business systems, and web services.
+- Delete the service/rc.local files or restore it to the original configuration.
+- Run a full antimalware scan. This may reveal additional artifacts left in the system, persistence mechanisms, and malware components.
+- Determine the initial vector abused by the attacker and take action to prevent reinfection through the same vector.
+- Leverage the incident response data and logging to improve the mean time to detect (MTTD) and the mean time to respond (MTTR).
+"""
+references = [
+    "https://www.intezer.com/blog/malware-analysis/hiddenwasp-malware-targeting-linux-systems/",
+    "https://pberba.github.io/security/2022/02/06/linux-threat-hunting-for-persistence-initialization-scripts-and-shell-configuration/#8-boot-or-logon-initialization-scripts-rc-scripts",
+    "https://www.cyberciti.biz/faq/how-to-enable-rc-local-shell-script-on-systemd-while-booting-linux-system/"
+
+]
+risk_score = 47
+rule_id = "0f4d35e4-925e-4959-ab24-911be207ee6f"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Investigation Guide"]
+type = "eql"
+
+query = '''
+sequence by user.id, host.id with maxspan=15s
+[file where event.type == "creation" and
+file.path : "/etc/rc.local"]
+[process where event.type == "start" and
+process.name == "chmod" and
+process.args : "+x" and
+process.args : "/etc/rc.local"]
```

Can be more precise with case sensitivity on Linux. Also, some spacing makes it slightly more readable and less error prone

```suggestion
sequence by user.id, host.id with maxspan=15s
[file where event.type == "creation" and
   file.path == "/etc/rc.local"]
[process where event.type == "start" and
   process.name == "chmod" and
   process.args == "+x" and process.args == "/etc/rc.local"]
```